### PR TITLE
Integrate model into app UI

### PR DESCRIPTION
**Pull request by angusYuhao**
1. Connected `runModel()` function with app UI
2. Graph and plot results from model

### CODE REVIEW

**Comment by liu-allan** [comment]
Application says it is not responding when running the model. Is this normal?

image

File: widget_pages/dashboard.py
Comment (liu-allan) [inline widget_pages/dashboard.py:296]: why are we running for `num_cycles + 1`?

**Review by liu-allan — COMMENTED** [review]
Perhaps we should have a confirmation box appear when the user wants to change pages. It takes a long time to load and I think we should make sure the user doesn't accidentally switch pages

**Review by liu-allan — COMMENTED** [review]
On line 73 in `dashboard.py` could you change the axis Neutrophil Count title to `Neutrophil Count (# Cells/L) x 1e9`.

File: widget_pages/dashboard.py
Comment (angusYuhao) [reply widget_pages/dashboard.py:296]: We are assuming that when the user enters 1 cycle to run, they want to run 1 cycle into the future. That means the model needs to run both the current cycle (for reference trajectory) and the future cycles -> 2 cycles to run

File: matlab_script.py
Comment (ianyechen) [reply matlab_script.py:33]: @liu-allan

**Review by ianyechen — COMMENTED** [review]
image
Not a big deal, but should we move the error message for the "number of calculation cycles" field being empty to like its actual box?

File: matlab_script.py
Comment (ianyechen) [inline matlab_script.py:27]: Don't forget to remove print statements when you're done :D

File: matlab_script.py
Comment (liu-allan) [reply matlab_script.py:33]: i can change this in the matlab

**Review by liu-allan — DISMISSED** [review]
Looks good, just remove the prints

File: widget_pages/dashboard.py
Comment (liu-allan) [reply widget_pages/dashboard.py:296]: Actually, I have more questions. If the patient has 3 measurements to use and you specify 2 cycles, does it calculate 5 cycles total? (3 for historical then 2 for the future?)

Comment (angusYuhao) [reply widget_pages/dashboard.py:296]: no, it should only run 3 cycles, the current one and 2 into the future

Comment (liu-allan) [reply widget_pages/dashboard.py:296]: So it only runs using the most recent measurement then?

**Comment by angusYuhao** [comment]
Should have addressed all comments and concerns except for the waiting screen when you press calculate, once you guys have reviewed it we can merge this. I will open a new PR for the waiting screen if we have time

File: widget_pages/dashboard.py
Comment (angusYuhao) [reply widget_pages/dashboard.py:296]: yeah, I believe so

**Review by liu-allan — APPROVED** [review]

**Comment by ianyechen** [comment]
Another minor comment:

image
What do you guys think about just completely removing the "Cancel" button. Right now it's in the same box as the calculate and it's kinda misleading. We have the toolbar on the left already... maybe that's enough? @angusYuhao @liu-allan @xieanne 

(This comment does not block the PR tho)

**Comment by ianyechen** [comment]
Also, have you decided on what to do in terms of saving the calculated data in memory yet? I think it would be nice to not have the model calculate again if nothing has changed? 

I guess it's not a big MUST to do. But would be nice.

**Comment by ianyechen** [comment]
image

When there's no historical data, and you do 1 calculation cycle. No output is expected right? Should we have some kind of prompt that tells the user this? Or maybe just stop them from calculating if there's no historical data?

**Review by ianyechen — APPROVED** [review]
LGTM, left some unblocking comments though.

**Review by xieanne — APPROVED** [review]
LGTM

**Pull request merged**
